src/unix.js

```diff
@@ -47,10 +47,11 @@ function escapeArgBash(arg, { interpolation, quoted }) {
   if (interpolation) {
     result = result
       .replace(/\\/gu, "\\\\")
-      .replace(/\n/gu, " ")
       .replace(/(^|\s)([#~])/gu, "$1\\$2")
       .replace(/(["$&'()*;<>?`{|])/gu, "\\$1")
-      .replace(/(?<=[:=])(~)(?=[\s+\-/0:=]|$)/gu, "\\$1");
+      .replace(/(?<=[:=])(~)(?=[\s+\-/0:=]|$)/gu, "\\$1")
+      .replace(/([\t ])/gu, "\\$1")
+      .replace(/\n/gu, " ");
```

Comment:
Consideration (applies to all Unix shells): escape `\n` (either as `\\\n` or `\\ `) to prevent argument splitting.

Reply:
See 4a0953dcb60f891a06c4b24623c9b2b7dacf576e